flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/MinAggFunctionWithRetract.scala

```diff
@@ -18,19 +18,20 @@
 package org.apache.flink.table.functions.aggfunctions
 
 import java.math.BigDecimal
-import java.util.{HashMap => JHashMap}
-import java.lang.{Iterable => JIterable}
+import java.lang.{Iterable => JIterable, Long => JLong}
 import java.sql.{Date, Time, Timestamp}
 
-import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, TypeInformation}
-import org.apache.flink.api.java.tuple.{Tuple2 => JTuple2}
-import org.apache.flink.api.java.typeutils.{MapTypeInfo, TupleTypeInfo}
-import org.apache.flink.table.api.Types
+import org.apache.flink.api.common.typeinfo.{BasicTypeInfo, TypeInformation, Types}
+import org.apache.flink.table.api.dataview.MapView
 import org.apache.flink.table.functions.aggfunctions.Ordering._
 import org.apache.flink.table.functions.AggregateFunction
 
 /** The initial accumulator for Min with retraction aggregate function */
-class MinWithRetractAccumulator[T] extends JTuple2[T, JHashMap[T, Long]]
+class MinWithRetractAccumulator[T] {
```

Comment:
I think we should override the `getAccumulatorType` for both `MinWithRetractAggFunction` and `MaxWithRetractAggFunction ` otherwise we will always use the memory.

Reply:
I think a better way to handle it is similar to: https://github.com/apache/flink/commit/dccdba199a8fbb8b5186f0952410c1b1b3dff14f. 
Just simply adding the mapView field to the class var argument.

Reply:
Thanks a lot for the suggestion. But I don't think the implementation of ACC type in CollectAggFunction is a better way. For example, it needs to define the `equals` and `canEqual` methods which are not needed if we use a POJO type instead.